opentelemetry/proto/profiles/v1experimental/profiles.proto

```diff
@@ -120,42 +129,143 @@ message ResourceProfiles {
-message ProfileContainer {
+// Profile is a common stacktrace profile format.
+//
+// Measurements represented with this format should follow the
+// following conventions:
+//
+// - Consumers should treat unset optional fields as if they had been
+//   set with their default value.
+//
+// - When possible, measurements should be stored in "unsampled" form
+//   that is most useful to humans.  There should be enough
+//   information present to determine the original sampled values.
+//
+// - On-disk, the serialized proto must be gzip-compressed.
+//
+// - The profile is represented as a set of samples, where each sample
+//   references a sequence of locations, and where each location belongs
+//   to a mapping.
+// - There is a N->1 relationship from sample.location_id entries to
+//   locations. For every sample.location_id entry there must be a
+//   unique Location with that index.
+// - There is an optional N->1 relationship from locations to
+//   mappings. For every nonzero Location.mapping_id there must be a
+//   unique Mapping with that index.
+
+// Represents a complete profile, including sample types, samples,
+// mappings to binaries, locations, functions, string table, and additional metadata.
+// It modifies and annotates pprof profile with OpenTelemetry specific fields.
+message Profile {
+
+  // first section: pprof fields.
+  // Note that whilst these retain the name and field id from pprof in most cases
+  // for ease of understanding data migration, it is not intended that pprof:Profile and
+  // OpenTelemetry:Profile encoding be wire compatible.
+
+  // A description of the samples associated with each Sample.value.
+  // For a cpu profile this might be:
+  //   [["cpu","nanoseconds"]] or [["wall","seconds"]] or [["syscall","count"]]
+  // For a heap profile, this might be:
+  //   [["allocations","count"], ["space","bytes"]],
+  // If one of the values represents the number of events represented
+  // by the sample, by convention it should be at index 0 and use
+  // sample_type.unit == "count".
+  repeated ValueType sample_type = 1;
+  // The set of samples recorded in this profile.
+  repeated Sample sample = 2;
+  // Mapping from address ranges to the image/binary/library mapped
+  // into that address range.  mapping_table[0] will be the main binary.
+  repeated Mapping mapping_table = 3;
+  // Locations referenced by samples via location_indices.
+  repeated Location location_table = 4;
+  // Functions referenced by locations.
+  repeated Function function_table = 5;
+  // A common table for strings referenced by various messages.
+  // string_table[0] must always be "".
+  repeated string string_table = 6;
+  // frames with Function.function_name fully matching the following
+  // regexp will be dropped from the samples, along with their successors.
+  int32 drop_frames_strindex = 7;   // Index into string table.
```

Comment:
I left this comment in the SIG meeting notes doc, copying here for visibility. Not directly related to this PR.

> Drop / keep frames regex: These fields are informal, just telling users what has been dropped/kept. I’d be okay with dropping them. Don’t think we would have designed this into the protocol.

In the pprof symbolization process these fields are not informal. The collectors populate them when producing the profile - the frames can't be dropped at that point since function names for native language like C++ are not known yet. Then the offline symbolization uses the values to actually drop some frames during the symbolization.

Basically these fields decouple the mechanism from the policy.

Reply:
ok. They stay for now anyhow. We'll take that into account when deciding if they stay longer term or get folded into attributes, which would still leave them present for use in this role.
On similar note I'm tempted to remove the deprecated fields in various places, but that may cause headaches for pprof data conversion if they have no suitable replacements.